envoy/router/router.h

```diff
@@ -53,9 +53,11 @@ class ResponseEntry {
    * example, adding or removing headers. This should only be called ONCE immediately after
    * obtaining the initial response headers.
    * @param headers supplies the response headers, which may be modified during this call.
+   * @param request_headers request headers to be used in manipulation of the response headers.
    * @param stream_info holds additional information about the request.
    */
   virtual void finalizeResponseHeaders(Http::ResponseHeaderMap& headers,
+                                       const Http::RequestHeaderMap& request_headers,
                                        const StreamInfo::StreamInfo& stream_info) const PURE;
```

Comment:
According to previous comment/discussion (https://github.com/envoyproxy/envoy/pull/21932#discussion_r965326784 and https://github.com/envoyproxy/envoy/pull/21932#issuecomment-1241188138), we will do some interface updates in a new PR. So, I am not sure if this change here is necessary for this PR? 

Could we defer this update to the PR that will unify all the interface (after the runtime guard is expired)?

Reply:
Remember that "new" (access log formatter) does not take `request_headers` from `stream_info` and they have to be passed to `format` method as parameter. The best we could do would be to get request headers from `stream_info` inside the `finalizeResponseHeaders`, so the signature stays the same. The thing is that each time `finalizeResponseHeaders` is called the request headers are easy to get, because they  are either local variables or class's date members.

Reply:
> The best we could do would be to get request headers from stream_info inside the finalizeResponseHeaders, so the signature stays the same.

This is what I want for this PR. If we will have a dedicated PR for interface updates, then I'd like delay all these interface updates to the future PR.

Reply:
Reverted to original signature and request headers are retrieved from `streamInfo` in `finalizeResponseHeaders` body.